### ISSUE TITLE
Fix numpy deprecation warnings

### ISSUE THREAD

**Issue by jfowkes** [fitbenchmarking/fitbenchmarking#824]
The latest version of numpy has deprecated the use of the np.* basic data types in favour of the built in python ones and raises a depreciation warning to that effect, as such we should make the following changes to our code:

```
np.bool -> bool
np.int -> int
np.float -> float
np.complex -> complex
np.object -> object 
np.str -> str
np.long -> int
np.unicode -> str
```

More details here: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

**Comment by tyronerees** [comment]
@jfowkes -- is this still an issue?  We think it might have been fixed (somehow) -- @robertapplin can't find these in the codebase

**Comment by jfowkes** [comment]
@tyronerees yes but it seems to be due to external solvers (bumps, dfogn, etc):
https://github.com/fitbenchmarking/fitbenchmarking/runs/3679098937

I guess we should raise this with the appropriate upstream?

**Comment by jess-farmer** [comment]
@RabiyaF to triage
